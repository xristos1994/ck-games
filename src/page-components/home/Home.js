import React from "react";
import { SEO, AvailableGames } from "./components";
import styles from "./styles.module.css";

const Home = () => {
  return (
    <div className={styles.pageBody}>
      <SEO />
      <AvailableGames />
    </div>
  );
};

export { Home };
