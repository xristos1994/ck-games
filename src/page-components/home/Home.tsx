import React, { FC, ReactElement, useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { SEO, AvailableGames } from './components';
import { compose } from '@utils/component-utils';
import { restartAllGames } from '@models/website/actions';
// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require('./Home.module.css');

interface IProps {
  restartAllGames: () => void;
}

export const _Home: FC<IProps> = ({ restartAllGames }): ReactElement => {
  useLayoutEffect(() => {
    restartAllGames();
  }, [restartAllGames]);

  return (
    <div className={styles.pageBody}>
      <SEO />
      <AvailableGames />
    </div>
  );
};

const Home = compose(connect(null, { restartAllGames }))(_Home);

export { Home };
