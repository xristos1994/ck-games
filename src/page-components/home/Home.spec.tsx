import React from 'react';
import renderer from 'react-test-renderer';
import { _Home as Home } from './Home';

const SEO = () => {
  return (
    <div>
      SEO
    </div>
  );
};

const AvailableGames = () => {
  return (
    <div>
      AvailableGames
    </div>
  );
};

jest.mock('./components/', () => ({
  __esModule: true,
  SEO: SEO,
  AvailableGames: AvailableGames
}));

const restartAllGames = jest.fn().mockImplementation(() => void 0);

describe('page-components/Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Home
        restartAllGames={restartAllGames}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
