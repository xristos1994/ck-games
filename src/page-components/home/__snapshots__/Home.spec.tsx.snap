// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/Home renders correctly 1`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    AvailableGames
  </div>
</div>
`;
